### PR TITLE
Switch to ESM module

### PR DESCRIPTION
**Pull request by Pessimistress**
For #2272
Targeting v7.2

### CODE REVIEW

File: test/src/components/map.spec.jsx [hunk 1]
```diff
@@ -107,14 +107,13 @@ test('Map#controlled#no-update', async t => {
   });
 });
 
-test('Map#controlled#mirrow-back', async t => {
-  t.plan(5);
+test('Map#controlled#mirror-back', async t => {
```
Comment (chrisgervang) [inline test/src/components/map.spec.jsx:110]: Fixing an unrelated typo?

File: test/src/utils/react-dom-mock.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import * as React from 'react';
+
+export function createPortal(element, container) {
```
Comment (chrisgervang) [inline test/src/utils/react-dom-mock.js:3]: Hoping to learn what this does. Where is this used now? I only see the usage getting deleted.

**Review by chrisgervang — APPROVED** [review]

File: test/src/utils/react-dom-mock.js (hunk 2 above)
Comment (Pessimistress) [reply test/src/utils/react-dom-mock.js:3]: This file is used as an alias for `react-dom` during node tests, configured in ocularrc. Before we were hijacking the function with `require('react-dom').createPortal = createPortalMock`, but that no longer works in the ESM (import/export) world.

Comment (chrisgervang) [reply test/src/utils/react-dom-mock.js:4]: Oh, ok got it. Could you comment that it's aliased in ocularrc?

**Pull request merged**